new-tutorial/3_message_passing.py

```diff
@@ -0,0 +1,353 @@
+"""
+Write your own GNN module
+=========================
+
+This tutorial assumes that you already know :doc:`the basics of training a
+GNN for node classification <1_introduction>`. However,
+sometimes you would like to invent a new way of aggregating neighbor
+information or a new way of using the aggregation.
+
+By the end of this tutorial you will be able to
+
+-  Understand DGL’s message passing APIs.
+-  Implement GraphSAGE convolution module by your own.
+
```

Comment:
Sometimes, your model goes beyond simply stacking existing GNN modules. For example, you would like to invent a new way of aggregating neighbor information by considering node importance or edge weights. By the end of this tutorial, you will be able to

-  Understand DGL’s message passing APIs.
-  Implement GraphSAGE convolution module by your own.

This tutorial requires knowledge of the basics of training a GNN for node classification <1_introduction>`.